### ISSUE TITLE
Gaps in sloping sides; extra perimeters insufficient

### ISSUE THREAD

**Issue by ericu** [alexrj/Slic3r#864]
This model has slopes at a variety of angles.  I set perimeters=2, generate extra perimeters when needed to true, and top solid layers to 3.  When I print, there are gaps in the slopes, as shown in the attached photo.

It looks to me like Slic3r is depending on perimeters to seal the top surface, and that the code that generates extra perimeters is failing.  So clearly there's a bug here, but I also think that it's going about it wrong.  If I set top_solid_layers to 3, I expect that, if I drill vertically down through the model /anywhere/, whether it be on a perfectly flat surface or a slant, I should have to go through at least 3 layers to get to the interior.  I shouldn't have to make the walls super-thick to make the top surface solid.  Slic3r should figure out where a layer is smaller than the one below, recognize the need for a top surface, and propagate that knowledge down by top_solid_layers to get the right minimum thickness.

My config, stl, gcode, and a photo are at https://docs.google.com/folder/d/0B1PhG4uhITMmbm9XWmJwVEFMMjA/edit

Here's the photo again, just for convenience:

![IMG_20121216_135439](https://f.cloud.github.com/assets/2245151/15817/ed00aede-47cc-11e2-8bd3-55cdd2959ae8.jpg)

I'm using 0.9.8-dev, built out of the git repo a couple of days ago, at commit cf8390b5d32ac1ae1afecdf01b9976e0316ae5c3, on Linux [Debian Squeeze].

You'll note that I printed this at pretty high speed; it didn't seem to make a difference at all when I slowed things down.

**Comment by lordofhyphens** [comment]
@ericu Can you try it with current 1.2.9?

**Comment by ericu** [comment]
Sorry @lordofhyphens, I haven't actually had my printer set up since I 1) moved; and 2) had a second kid.  Maybe in another year, but I'm kind of doubtful.

**Comment by lordofhyphens** [comment]
@ericu your model has since disappeared off the interwebs, could you zip it up and drop it here if you can find it?

**Comment by ericu** [comment]
I don't have that computer handy, but I see that the version that mesheldrake posted above is still a valid link: http://sheldrake.net/bag/moebius_slice_scaled.stl

**Comment by djdelorie** [comment]
I tried to reproduce this today, adding my observations (for better or worse ;) :
since I couldn't get the original model via google docs, I re-implemented one: http://www.delorie.com/tmp/mobius.scad
Printed with slic3r 1.3.0-dev, it looks like slic3r is trying to put in enough layers, but tries to bridge to open air: http://www.delorie.com/tmp/mobius-bridge.png
Resulting in gaps due to non-adhesion: http://www.delorie.com/photos/3d_prints/img_3246.html

**Comment by DrLex0** [comment]
@djdelorie Yes, lack of proper anchoring of the first layer on top of infill is a known issue: #1815
